internal/server/server_test.go

```diff
@@ -0,0 +1,74 @@
+package server
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestParseMatrixDimensions(t *testing.T) {
+
+	data := []struct {
+		name    string
+		vars    map[string][]string
+		rows    int
+		columns int
+		err     error
+	}{
+		{
+			name:    "Missing Row Parameter",
+			vars:    map[string][]string{"columns": {"2"}},
+			rows:    0,
+			columns: 0,
+			err:     errors.New("Please enter both row and column parameters and values"),
+		},
+		{
+			name:    "Missing Column Parameter",
+			vars:    map[string][]string{"rows": {"d"}},
+			rows:    0,
+			columns: 0,
+			err:     errors.New("Please enter both row and column parameters and values"),
+		},
+		{
+			name:    "Row Invalid",
+			vars:    map[string][]string{"rows": {"d"}, "columns": {"2"}},
+			rows:    0,
+			columns: 0,
+			err:     errors.New("Please enter an *integer* for your desired number of rows"),
+		},
+		{
+			name:    "Column Invalid",
+			vars:    map[string][]string{"rows": {"7"}, "columns": {"f"}},
+			rows:    0,
+			columns: 0,
+			err:     errors.New("Please enter an *integer* for your desired number of columns"),
+		},
+		{
+			name:    "All Valid",
+			vars:    map[string][]string{"rows": {"3"}, "columns": {"5"}},
+			rows:    3,
+			columns: 5,
+			err:     nil,
+		},
+	}
+
+	for _, d := range data {
+		t.Run(d.name, func(t *testing.T) {
+			rows, columns, err := ParseMatrixDimensions(d.vars)
+			if rows != d.rows {
+				t.Errorf("Expected %d, got %d", d.rows, rows)
+			}
+			if columns != d.columns {
+				t.Errorf("Expected %d, got %d", d.columns, columns)
+			}
+			if err == nil && d.err != nil {
+				t.Errorf("Expected %d, got %d", d.err, err)
+			} else if err != nil && d.err == nil {
+				t.Errorf("Expected %d, got %d", d.err, err)
+			} else if err != nil && d.err != nil {
+				if err.Error() != d.err.Error() {
+					t.Errorf("Expected %d, got %d", d.err, err)
+				}
+			}
```

Comment:
This is fine, when there's not a better way of testing. But you should look into using the require package to improve the readability of your tests https://pkg.go.dev/github.com/stretchr/testify/require

Reply:
@AYM1607 I used the require package and my tests are passing, but could I get your eyes to make sure the changes I made are what you had in mind? Thanks a bunch!

Reply:
It is very simple! Testing frameworks make a huge difference when writing expectations. I left a few comments, will resolve this since the package was already added.